lib/timber/events/custom_event.ex

```diff
@@ -36,17 +58,26 @@ defmodule Timber.Events.CustomEvent do
     data: map() | nil
   }
 
+  defmacro __using__(opts) do
```

Comment:
You know, I keep going back and forth on this. My case for using a protocol is the ability to map external data structures to a `Timber.Event.t`, including primitive types like `Map`. This allows users to do:

```elixir
Logger.info "Payment rejected", event: %{type: :payment_rejected, data: %{...}}
Logger.info "Payment rejected", event: %PaymentRejectedEvent{...}
```

I'm not a fan of the following because there is no way to structure the data passed to `CustomEvent`:

```elixir
Logger.info "Payment rejected", event: %Timber.Events.CustomEvent{type: :payment_rejected, data: %{...}}
```

How would you accomplish the second line in my first example without a protocol?